crates/red_knot_python_semantic/src/semantic_index/attribute_assignment.rs

```diff
@@ -14,6 +17,17 @@ pub(crate) enum AttributeAssignment<'db> {
 
     /// An attribute assignment without a type annotation, e.g. `self.x = <value>`.
     Unannotated { value: Expression<'db> },
+
+    /// An attribute assignment where the right-hand side is an iterable, for example
+    /// `for self.x in <iterable>`.
+    Iterable { iterable: Expression<'db> },
+
+    /// An attribute assignment where the left-hand side is an unpacking expression,
+    /// e.g. `self.x, self.y = <value>`.
+    Unpack {
+        attribute_expression_id: ScopedExpressionId,
+        unpack: Unpack<'db>,
+    },
```

Comment:
With the new variants, this now looks similar to `CurrentAssignment`, but there are differences, and also, attribute assignments are not supported in all scenarios. For example, the target in a *name*d expression can only be a … *name*! So something like `(self.x := 1)` is not legal.